src/bit-systems/scene-preview-camera-system.ts

```diff
@@ -0,0 +1,119 @@
+import { defineQuery, enterQuery, exitQuery } from "bitecs";
+import { HubsWorld } from "../app";
+import { ScenePreviewCamera } from "../bit-components";
+import { Quaternion, Vector3, Euler, Camera } from "three";
+import { CAMERA_MODE_INSPECT, CAMERA_MODE_SCENE_PREVIEW, CameraSystem } from "../systems/camera-system";
+import { setMatrixWorld } from "../utils/three-utils";
+import { getStreamerCamera } from "../components/scene-preview-camera";
+import { lerp } from "three/src/math/MathUtils";
+
+const startPositions = new Map<number, Vector3>();
+const startRotations = new Map<number, Quaternion>();
+const targetPositions = new Map<number, Vector3>();
+const targetRotations = new Map<number, Quaternion>();
+const startTimes = new Map<number, number>();
+const backwards = new Map<number, boolean>();
+const ranOnePass = new Map<number, boolean>();
+const newRot = new Quaternion();
```

Comment:
This comment doesn't block the PR but is for future discussion.

Perhaps it would be good if we can define a standard pattern about where working variables (or states) required by a system are stored for the consistency. Ex: In a system file, in component, or another working data component. I added this topic to the internal design document.